Adding State Management Options to CLI

theblondealex opened the pull request:
## Description

I had added state management options to the ces flow
- Initially I have only added Zustand but with the approval of this PR I will look into adding other optons too
- Adds a question on the flow (see image) 
- Adds a --zustand flag to skip the question too.
This creates a folder in the root directory called state management with a default store from the zustand docs, it could be updated in future with a more generic store and some comments on how to use but to start with its the default store from zustand docs

## Motivation and Context

My motivation was as I believe that ces can and should become one of the best and number one go to option for starting new RN expo apps and state management is mandatory in almost every app asit scales and having the setup from the get go is a must, as well as this the intuitive nature of ces means that for beginner RN devs they can use it and this will ease them into state management, in short ces should have eveything you could need, state management is one such item

## Screenshots (if appropriate):

![image](https://github.com/danstepanov/create-expo-stack/assets/65453441/160a6501-65ce-4a42-a325-9f4cde4704ec)

danstepanov commented:
@theblondealex planning to review this for the next version of CES (next week)

Avaray commented:
I really wish to see this in the CLI

theblondealex commented:
Just updated it to resolve some conflicts with the recent main branch push

theblondealex commented on cli/src/utilities/runCLI.ts:
```diff
@@ -404,6 +405,37 @@ export async function runCLI(toolbox: Toolbox, projectName: string): Promise<Cli
+      message: 'What would you like to use for state management?',
+      options: [
+        { value: undefined, label: 'None' },
+        { value: 'zustand', label: 'Zustand' }
```
re the first comment about flag as there is another [PR ](https://github.com/roninoss/create-expo-stack/pull/417) up for redux might be good to keep as an option and allow it there

Apologies for such a delay got distrated with the day job, here is the video you requested

*few out of date package warnings but that will be solved with a rebase to main/merge* can rebase if you'd like but thought get approval on it first than to waste time if other changes need to be made


https://github.com/user-attachments/assets/30c95f01-57fc-4126-a2c9-78f98b808b02

dannyhw replied:
agree, lets add both redux and zustand

theblondealex commented:
> Have you tested this while also using 'react-native-mmkv' ? Having a hard time finding examples of this setup and even more so if including @TanStack-Query.

Those are separate packages to zustand and aren't exactly state managers so most likely wouldnt be in here IMO, this is the Redux/zustand type packages

dannyhw commented:
Hey sorry that its taken a while but I will be looking to get things moving on the state management templates and will be taking a look soon.

dannyhw commented on cli/src/utilities/runCLI.ts:
```diff
@@ -384,6 +385,37 @@ export async function runCLI(toolbox: Toolbox, projectName: string): Promise<Cli
+      success(`No problem, skipping state management for now.`);
+    }
+
+    const internationalizationSelect = await confirm({
```
not sure this internationalisation thing is doing anything here

dannyhw commented:
There were some weird things that seem to have been messed up from merge conflicts probably, I think i managed to resolve them though

theblondealex commented:
> There were some weird things that seem to have been messed up from merge conflicts probably, I think i managed to resolve them though

Awesome does anything else need to be done? Can I help in any way?

dannyhw commented:
I think we're all good, just made a few more adjustments

dannyhw commented:
@danstepanov your change request is blocking merge could you take a look, I think we can go ahead with this as is. I'll merge the redux changes in basically right after this is merged.

danstepanov reviewed (approved):
LGTM

dannyhw commented:
@theblondealex we're having trouble with rebasing this branch, looks like the commit history got a bit messed up.

Can you try rebasing against main?

theblondealex commented:
@dannyhw I have squashed the commits and rebased it against the latest, this inadvertedly closed the PR so reopened it now and it should be ready to merge

dannyhw commented:
just ran everything locally and looks like we're good
